server/models/contests/participate.js

```diff
@@ -0,0 +1,77 @@
+const { pool } = require('../database')
+
+/**
+ * @param {*} param0
+ * @param {Object} param0.params
+ * @param {String} param0.username
+ * @return {Promise}
+ */
+
+function participate({ params, username }) {
+  return new Promise((resolve, reject) => {
+    const { contest_id: contestId } = params
+    pool.getConnection((error, connection) => {
+      if (error) {
+        return reject(error)
+      }
+      connection.beginTransaction((error) => {
+        if (error) {
+          connection.release()
+          return reject(error)
+        }
+        connection.query(
+          `INSERT INTO contests_participants (contest_id, participant) SELECT ?, ? 
+          WHERE EXISTS(SELECT 1 FROM user_groups ug INNER JOIN contests_groups cg ON ug.group_id = cg.group_id WHERE ug.username=? AND cg.contest_id=?)`,
+          [contestId, username, username, contestId],
+          (error, res) => {
+            if (error || res === undefined) {
+              return connection.rollback(() => {
+                connection.release()
+                const { code } = error
+                if (code === 'ER_DUP_ENTRY') {
+                  return reject('The user is already a participant')
+                }
+                return reject(error)
+              })
+            }
+            const { affectedRows } = res
+            if (!affectedRows) {
+              return connection.rollback(() => {
+                connection.release()
+                return reject(
+                  'None of the groups user is in are eligible for the contest'
```

Comment:
Easy to understand for the user, since this message would be directly displayed in frontend-
```suggestion
                  'You are not eligible to register for the contest'
```